### ISSUE TITLE
cdc tidb changefeed checkpoint cannot advance 

### ISSUE THREAD

**Issue by AkiraXie** [pingcap/tiflow#7766]
### What did you do?

1. tidb changefeed running
2. 11/30 create storage changefeed
3. after creating storage changefeed , ticdc meeted panic 
4. upgrade ticdc , ticdc won't panic
5. storage changefeed remove

### What did you expect to see?

tidb changefeed lag can be decrease after upgrade

### What did you see instead?

1. changefeed lag is increasing before upgrade
![image](https://user-images.githubusercontent.com/50363732/205019997-2441b043-ff95-4014-8ecf-ae75ec9a95d6.png)
2.  changefeed lag is increasing after upgrade
![image](https://user-images.githubusercontent.com/50363732/205020493-e4462bda-e5c3-474f-aacb-889cafb0f201.png)

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):
Release Version: v6.5.0-alpha
Edition: Community
Git Commit Hash: cca78e37e249c235c9664fc6b5f6c2ac4b4a3b79
Git Branch: heads/refs/tags/v6.5.0-alpha
UTC Build Time: 2022-11-30 00:24:59
GoVersion: go1.19.3
Race Enabled: false
TiKV Min Version: 6.2.0-alpha
Check Table Before Drop: false
Store: unistore

```console
(paste TiDB cluster version here)
```

Upstream TiKV version (execute `tikv-server --version`):

```console
(paste TiKV version here)
```
TiKV 
Release Version:   6.5.0-alpha
Edition:           Community
Git Commit Hash:   bfe29655a2a1634c268a7fb3986892be7dc1f305
Git Commit Branch: heads/refs/tags/v6.5.0-alpha
UTC Build Time:    2022-11-28 11:07:57
Rust Version:      rustc 1.67.0-nightly (96ddd32c4 2022-11-14)
Enable Features:   pprof-fp jemalloc mem-profiling portable sse test-engine-kv-rocksdb test-engine-raft-raft-engine cloud-aws cloud-gcp cloud-azure
Profile:           dist_release

TiCDC version (execute `cdc version`):
before update:
Release Version: v6.5.0-alpha
Git Commit Hash: 07657c5fa7035f4dc02f82473c50d516ff7b58ff
Git Branch: heads/refs/tags/v6.5.0-alpha
UTC Build Time: 2022-11-28 11:06:34
Go Version: go version go1.19.3 linux/amd64
Failpoint Build: false
now:
Release Version: v6.5.0-alpha
Git Commit Hash: 519c6f08dc5ca000d8c92101e2987daf071d716e
Git Branch: heads/refs/tags/v6.5.0-alpha
UTC Build Time: 2022-11-30 11:07:02
Go Version: go version go1.19.3 linux/amd64
Failpoint Build: false

**Comment by fubinzh** [comment]
This issue is also seen when create storage sink with NFS storage target.
```
[2022/12/02 15:45:52.108 +08:00] [WARN] [replication_manager.go:499] ["schedulerv3: cannot advance checkpoint since missing table"] [namespace=default] [changefeed=test2] [tableID=78]
[2022/12/02 15:45:52.157 +08:00] [WARN] [replication_manager.go:499] ["schedulerv3: cannot advance checkpoint since missing table"] [namespace=default] [changefeed=test2] [tableID=78]
[2022/12/02 15:45:52.168 +08:00] [WARN] [client.go:259] ["etcd client outCh blocking too long, the etcdWorker may be stuck"] [duration=59m48.001033165s] [role=processor]
[2022/12/02 15:45:52.207 +08:00] [WARN] [replication_manager.go:499] ["schedulerv3: cannot advance checkpoint since missing table"] [namespace=default] [changefeed=test2] [tableID=78]
```

**Comment by zhaoxinyu** [comment]
Maybe fixed by #7777.

**Comment by fubinzh** [comment]
> This issue is also seen when create storage sink with NFS storage target.
> 
> ```
> [2022/12/02 15:45:52.108 +08:00] [WARN] [replication_manager.go:499] ["schedulerv3: cannot advance checkpoint since missing table"] [namespace=default] [changefeed=test2] [tableID=78]
> [2022/12/02 15:45:52.157 +08:00] [WARN] [replication_manager.go:499] ["schedulerv3: cannot advance checkpoint since missing table"] [namespace=default] [changefeed=test2] [tableID=78]
> [2022/12/02 15:45:52.168 +08:00] [WARN] [client.go:259] ["etcd client outCh blocking too long, the etcdWorker may be stuck"] [duration=59m48.001033165s] [role=processor]
> [2022/12/02 15:45:52.207 +08:00] [WARN] [replication_manager.go:499] ["schedulerv3: cannot advance checkpoint since missing table"] [namespace=default] [changefeed=test2] [tableID=78]
> ```

Not see this issue with latest build

**Comment by jebter** [comment]
https://github.com/pingcap/tiflow/issues/7780

**Comment by kennytm** [comment]
> This issue is also seen when create storage sink with NFS storage target.
> 
> ```
> [2022/12/02 15:45:52.108 +08:00] [WARN] [replication_manager.go:499] ["schedulerv3: cannot advance checkpoint since missing table"] [namespace=default] [changefeed=test2] [tableID=78]
> [2022/12/02 15:45:52.157 +08:00] [WARN] [replication_manager.go:499] ["schedulerv3: cannot advance checkpoint since missing table"] [namespace=default] [changefeed=test2] [tableID=78]
> [2022/12/02 15:45:52.168 +08:00] [WARN] [client.go:259] ["etcd client outCh blocking too long, the etcdWorker may be stuck"] [duration=59m48.001033165s] [role=processor]
> [2022/12/02 15:45:52.207 +08:00] [WARN] [replication_manager.go:499] ["schedulerv3: cannot advance checkpoint since missing table"] [namespace=default] [changefeed=test2] [tableID=78]
> ```

We have recently see this issue again on v6.5.2 with an *extremely faulty NFS drive* (returning "invalid argument" error when closing files). The "missing table" error goes on for 4 hours since the changefeed inititialized until CDC restarted. This seems probabilistic (sometimes works sometimes stuck) and irrelevant to closing the sink manager.

Should we file a new issue for this?

**Comment by CharlesCheung96** [comment]
@kennytm Could you please provide the log file of ticdc?

**Comment by kennytm** [comment]
(sent privately)

**Comment by CharlesCheung96** [comment]
@kennytm 
1. When changefeed is restarted or created, the owner scheduler [sends heartbeat information to all the capture nodes](https://github.com/pingcap/tiflow/blob/v6.5.2/cdc/scheduler/internal/v3/coordinator.go#L270) and determines that the schduler initialization is successful [only after receiving replies from all nodes](https://github.com/pingcap/tiflow/blob/master/cdc/scheduler/internal/v3/member/capture_manager.go#L152-L166).
2. The logs show that one of the processors in the cluster has been stuck, and some logs indicate that the p2p system is dropping messages due to congestion:
```log
[2023/07/03 14:24:20.210 +08:00] [WARN] [client.go:171] ["peer message client detected error, restarting"] [error="[CDC:ErrPeerMessageServerClosed]peer-to-peer message server has closed connection: [CDC:ErrPeerMessageTopicCongested]peer-to-peer message topic has congested, aborting all connections."] [errorVerbose="[CDC:ErrPeerMessageServerClosed]peer-to-peer message server has closed connection: [CDC:ErrPeerMessageTopicCongested]peer-to-peer message topic has congested, aborting all connections.\ngithub.com/pingcap/errors.AddStack\n\tgithub.com/pingcap/errors@v0.11.5-0.20220729040631-518f63d66278/errors.go:174\ngithub.com/pingcap/errors.(*Error).GenWithStackByArgs\n\tgithub.com/pingcap/errors@v0.11.5-0.20220729040631-518f63d66278/normalize.go:164\ngithub.com/pingcap/tiflow/pkg/p2p.(*MessageClient).runRx\n\tgithub.com/pingcap/tiflow/pkg/p2p/client.go:360\ngithub.com/pingcap/tiflow/pkg/p2p.(*MessageClient).run.func2\n\tgithub.com/pingcap/tiflow/pkg/p2p/client.go:208\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.1.0/errgroup/errgroup.go:75\nruntime.goexit\n\truntime/asm_amd64.s:1594"]
[2023/07/03 14:24:20.910 +08:00] [WARN] [client.go:259] ["etcd client outCh blocking too long, the etcdWorker may be stuck"] [duration=47h37m5.499571149s] [role=processor]
```
3. In addition, between `2023/07/03 10:24:42.557` and `2023/07/03 14:23:26.207,` no logs were found indicating that [all capture initializations were successful](https://github.com/pingcap/tiflow/blob/master/cdc/scheduler/internal/v3/member/capture_manager.go#L278-L279), which confirms the analysis of the second point above.
4. Eventually [the scheduler gets stuck in an uninitialized state](https://github.com/pingcap/tiflow/blob/v6.5.2/cdc/scheduler/internal/v3/coordinator.go#L289-L294), and never [performed a scheduling task](https://github.com/pingcap/tiflow/blob/v6.5.2/cdc/scheduler/internal/v3/coordinator.go#L309-L313), which caused the `missing table` error.

In conclusion, to solve this problem, we need to figure out why the processor is stuck.

**Comment by kennytm** [comment]
The issue above has been identified as #9370.
